server/apps/immich/src/api-v1/album/dto/create-album-shared-link.dto.ts

```diff
@@ -0,0 +1,19 @@
+import { IsBoolean, IsNotEmpty, IsOptional, IsString } from 'class-validator';
+
+export class CreateAlbumShareLinkDto {
+  @IsString()
+  @IsNotEmpty()
+  albumId!: string;
+
+  @IsString()
+  @IsOptional()
+  expiredAt?: string;
+
+  @IsBoolean()
+  @IsOptional()
+  allowUpload?: boolean;
+
+  @IsString()
+  @IsOptional()
+  description?: string;
```

Comment:
These are probably shared with the other create album dto, so it would be nice if they inherited/extended each other. Like have a base class and then two different sub classes.